lib/wallet/http.js

```diff
@@ -833,6 +783,405 @@ class HTTP extends Server {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+      let own = valid.bool('own', false);
+
+      if (name)
+        assert(rules.verifyName(name), 'Must pass valid name.');
+
+      if (!name)
+        own = true;
+
+      const reveals = await req.wallet.getRevealsByName(name);
+      const items = [];
+
+      for (const brv of reveals) {
+        if (!own || brv.own)
+          items.push(brv.toJSON());
+      }
+
+      res.json(200, items);
+    });
+
+    // Name Resource
+    this.get('/wallet/:id/resource', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+
+      assert(name, 'Must pass name.');
+      assert(rules.verifyName(name), 'Must pass valid name.');
+
+      const ns = await req.wallet.getNameStateByName(name);
+
+      if (!ns || ns.data.length === 0)
+        return res.json(404);
+
+      try {
+        const resource = Resource.decode(ns.data);
+        return res.json(200, resource.toJSON());
+      } catch (e) {
+        return res.json(400);
+      }
+    });
+
+    // Create Open
+    this.post('/wallet/:id/open', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+      const force = valid.bool('force', false);
+      const passphrase = valid.str('passphrase');
+      const broadcast = valid.bool('broadcast', true);
+      const sign = valid.bool('sign', true);
+
+      assert(name, 'Name is required.');
+      assert(broadcast ? sign : true, 'Must sign when broadcasting.');
+
+      const options = TransactionOptions.fromValidator(valid);
+      const mtx = await req.wallet.createOpen(name, force, options);
+
+      if (broadcast) {
+        const tx = await req.wallet.sendMTX(mtx, passphrase);
+        return res.json(200, tx.getJSON(this.network));
+      }
+
+      if (sign)
+        await req.wallet.sign(mtx, passphrase);
+
+      return res.json(200, mtx.getJSON(this.network));
+    });
+
+    // Create Bid
+    this.post('/wallet/:id/bid', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const name = valid.str('name');
+      const bid = valid.u64('bid');
+      const lockup = valid.u64('lockup');
+      const passphrase = valid.str('passphrase');
+      const broadcast = valid.bool('broadcast', true);
+      const sign = valid.bool('sign', true);
+
+      assert(name, 'Name is required.');
+      assert(bid, 'Bid is required.');
+      assert(lockup, 'Lockup is required.');
+      assert(broadcast ? sign : true, 'Must sign when broadcasting.');
+
+      const options = TransactionOptions.fromValidator(valid);
+      const mtx = await req.wallet.createBid(name, bid, lockup, options);
+
+      if (broadcast) {
+        const tx = await req.wallet.sendMTX(mtx, passphrase);
+        return res.json(200, tx.getJSON(this.network));
+      }
+
+      if (sign)
+        await req.wallet.sign(mtx, passphrase);
```

Comment:
Should we sign before broadcast? :-P Won't this return before signing?

Reply:
`wallet.sendMTX` signs the transaction, which isn't immediately clear just by looking at this file.
https://github.com/handshake-org/hsd/blob/41ff9172ba32ce4b2af78dc11ec4045e0b37168e/lib/wallet/wallet.js#L3255

It was designed so that you could:

- sign and broadcast
- sign and not broadcast
- not sign and not broadcast

I believe this would allow for maximum flexibility when it comes to using `hsd` with watch only wallets and payment channels.